api/service.go

```diff
@@ -621,15 +777,22 @@ func (s *service) AddFile(server pb.API_AddFileServer) error {
 	pth, err := s.ipfsClient.Unixfs().Add(
 		server.Context(),
 		ipfsfiles.NewReaderFile(reader),
-		options.Unixfs.Pin(true),
+		options.Unixfs.Pin(false),
```

Comment:
Uhm, so the root-folder was pinned at creation but a priori here isn't pinned.

Reply:
This just means, don't create an _additional_ pin for the actual file since below we're going to place a pin above it at the new folder root.

Reply:
So to clarify: pinning the base directory is recursive.

Reply:
Yep. To further add confusion, the pins under the root are not normal type pins... they are _recursive_ type pins, which makes them easy to remove when you remove the root pin.